src/backend/gpopt/gpdbwrappers.cpp

```diff
@@ -2617,6 +2610,13 @@ gpdb::MDCacheNeedsReset(void)
 			last_mdcache_invalidation_counter = mdcache_invalidation_counter;
 			return true;
 		}
+
+		if ((gpdb::GPDBTransactionIdIsValid(
```

Comment:
Now, if code on 2607 line returns false, then your new logic can not return true if need

Reply:
What's the problem to introduce the extra bool variable and make multiple assigments and one return?

Reply:
or simple inject new logic before returning false

Reply:
```diff
--- return false;
+++ return (gpdb::GPDBTransactionIdIsValid(gpopt::CMDCache::GetTransientXmin()) && gpdb::GetTransactionXmin() != gpopt::CMDCache::GetTransientXmin()))
```

Reply:
ok, but I would make a separate if to make it clear that this is a separate logic.

Reply:
> ok, but I would make a separate if to make it clear that this is a separate logic.

Yes, it would be more readable

Reply:
And add the appropriate comments. Both above the function signature and new code